awx/main/models/credential/injectors.py

```diff
@@ -32,21 +35,25 @@ def gce(cred, env, private_data_dir):
 
 
 def azure_rm(cred, env, private_data_dir):
-    if len(cred.client) and len(cred.tenant):
-        env['AZURE_CLIENT_ID'] = cred.client
-        env['AZURE_SECRET'] = decrypt_field(cred, 'secret')
-        env['AZURE_TENANT'] = cred.tenant
-        env['AZURE_SUBSCRIPTION_ID'] = cred.subscription
+    client = cred.get_input('client')
+    tenant = cred.get_input('tenant')
+
+    if len(client) and len(tenant):
+        env['AZURE_CLIENT_ID'] = client
+        env['AZURE_TENANT'] = tenant
+        env['AZURE_SECRET'] = cred.get_input('secret')
+        env['AZURE_SUBSCRIPTION_ID'] = cred.get_input('subscription')
```

Comment:
There's no reason to keep `AZURE_SUBSCRIPTION_ID` in the `if` block, because it's done in both cases. I think it will be more clear to move it before the `if`

Reply:
I agree with you, but the `if` logic already exists on devel https://github.com/ansible/awx/blob/devel/awx/main/models/credential/injectors.py#L35 so I don't want to replace it in this PR, as the goal of this PR is to not change any functionality.